src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -108,40 +109,40 @@ object KustoDataSourceUtils{
     var keyVaultAuthentication: Option[KeyVaultAuthentication] = None
     if(keyVaultUri != ""){
       // KeyVault Authentication
-      var keyVaultAppId: String = ""
+      val keyVaultAppId: String = parameters.getOrElse(KustoOptions.KEY_VAULT_APP_ID, "")
 
-      if({keyVaultAppId = parameters.getOrElse(KustoOptions.KEY_VAULT_APP_ID, "")
-        keyVaultAppId != ""}){
+      if(!keyVaultAppId.isEmpty){
         keyVaultAuthentication = Some(KeyVaultAppAuthentication(keyVaultUri,
           keyVaultAppId,
           parameters.getOrElse(KustoOptions.KEY_VAULT_APP_KEY, "")))
-      }
-      else {
+      } else {
         keyVaultAuthentication = Some(KeyVaultCertificateAuthentication(keyVaultUri,
           parameters.getOrElse(KustoOptions.KEY_VAULT_PEM_FILE_PATH, ""),
           parameters.getOrElse(KustoOptions.KEY_VAULT_CERTIFICATE_KEY, "")))
       }
     }
 
-    if(applicationId != "" || applicationKey != ""){
+    if(!applicationId.isEmpty || !applicationKey.isEmpty){
```

Comment:
applicationId.isDefined || applicationKey.isDefined

Reply:
Don't you actually need applicationId.isDefined **&&** applicationKey.isDefined here?